sdk/storage/Azure.Storage.Webjobs.Extensions.Blobs/src/BlobNameValidationAttribute.cs

```diff
@@ -20,7 +20,7 @@ namespace Microsoft.Azure.WebJobs.Description
     /// Validate this on the client side so that we can get a user-friendly error rather than a 400 from the service.
     /// See code here: http://social.msdn.microsoft.com/Forums/en-GB/windowsazuredata/thread/d364761b-6d9d-4c15-8353-46c6719a3392
     /// </summary>
-    public class BlobNameValidationAttribute : ValidationAttribute
+    internal sealed class BlobNameValidationAttribute : ValidationAttribute
```

Comment:
Right - because where these are validated https://github.com/Azure/azure-webjobs-sdk/blob/ccc20de1fb7fd7043aed6d12a2c46acce1d7a49a/src/Microsoft.Azure.WebJobs.Host/Bindings/AttributeCloner.cs#L199 we look for ValidationAttribute.

Reply:
Did you not carry the PublicSurface tests forward to this new repo? https://github.com/Azure/azure-webjobs-sdk/blob/5e410d7ae1c7c1b54b81b6a50faa5cc226deba5e/test/Microsoft.Azure.Webjobs.Extensions.Storage.UnitTests/PublicSurfaceTests.cs#L55. We had a set of useful tests that helped us verify public surface area.

Reply:
I did initially. However the `azure-sdk-for-net` has an infrastructure to extract public surface into files like `/api/Azure.WebJobs.Extensions.Storage.Blobs.netstandard2.0.cs` (included in this diff) and warn against public area changes.
There's also a process to review public surface area before shipping any public preview or GA which is based on these extracts.

Reply:
Here's more about the process https://github.com/Azure/azure-sdk/blob/master/docs/policies/reviewprocess.md

Reply:
👍

Reply:
Sounds good, thanks